src/main/java/tool/designpatterns/verifiers/multiclassverifiers/proxy/ProxyFeedbackUtils.java

```diff
@@ -0,0 +1,373 @@
+
+        return newPartial;
+    }
+
+    /**
+     * Splits the given list in valid and invalid pattern groups.
+     *
+     * @param patternGroups the pattern groups to split.
+     *
+     * @return a tuple containing two new lists, the first containing only the valid
+     *     ProxyPatternGroups in the original list and the second containing only the invalid ones.
+     */
+    private static Tuple2<List<ProxyPatternGroup>, List<ProxyPatternGroup>> splitToValidInvalid(
+        List<ProxyPatternGroup> patternGroups) {
+        List<ProxyPatternGroup> invalid = new ArrayList<>();
+        List<ProxyPatternGroup> valid = new ArrayList<>();
+
+        for (ProxyPatternGroup group : patternGroups) {
+            if (group.isValid()) {
+                valid.add(group);
+            } else {
+                invalid.add(group);
+            }
+        }
+        return new Tuple2<>(valid, invalid);
+    }
+
+    /**
+     * Returns the feedbacks from the invalid ProxyPatternGroups.
+     *
+     * @param invalid the invalid ProxyPatternGroups.
+     * @param valid   the valid ProxyPatternGroupos.
+     *
+     * @return the feedbacks from the given ProxyPatternGroups.
+     */
+    private static List<Feedback> getInvalidFeedback(
+        List<ProxyPatternGroup> invalid, List<ProxyPatternGroup> valid) {
+        List<Feedback> feedbacks = new ArrayList<>();
+
+        for (ProxyPatternGroup invalidGroup : invalid) {
+            // Must compare the qualified names, otherwise there can be issues with identical
+            // classes in different packages.
+            String invalidProxyName = invalidGroup.getProxy().resolve().getQualifiedName();
+            String invalidInterfName = invalidGroup.getInterfaceOrAClass().resolve()
+                                                   .getQualifiedName();
+            String invalidSubjectName = invalidGroup.getSubject().resolve().getQualifiedName();
+
+            // Add the feedback if one or more of the classes aren't used in a valid PatternGroup.
+            boolean proxyUsed = false;
+            boolean interfaceUsed = false;
+            boolean subjectUsed = false;
+
+            for (ProxyPatternGroup validGroup : valid) {
+                String validProxyQualName = validGroup.getProxy().resolve().getQualifiedName();
+                String validInterfQualName = validGroup.getInterfaceOrAClass().resolve()
+                                                       .getQualifiedName();
+                String validSubjectQualName = validGroup.getSubject().resolve().getQualifiedName();
+
+                if (validProxyQualName.equals(invalidProxyName)) {
+                    proxyUsed = true;
+                }
+
+                if (validInterfQualName.equals(invalidInterfName)) {
+                    interfaceUsed = true;
+                }
+
+                if (validSubjectQualName.equals(invalidSubjectName)) {
+                    subjectUsed = true;
+                }
+            }
```

Comment:
couldn't the helper method isSameClassIsh method in ClassVerification be used here?

Reply:
Solved in b6f9710